packages/field-base/src/label-controller.js

```diff
@@ -0,0 +1,216 @@
+
+        this.__observeLabel(node);
+      }
+    );
+  }
+
+  /** @protected */
+  get uniqueId() {
+    return SlotController.labelId;
+  }
+
+  /**
+   * Override to initialize the newly added custom label.
+   *
+   * @param {Node} labelNode
+   * @protected
+   * @override
+   */
+  initCustomNode(labelNode) {
+    this.__applyCustomLabel(labelNode);
+
+    this.__observeLabel(labelNode);
+  }
+
+  /**
+   * Override to cleanup label node when it's removed.
+   *
+   * @param {Node} node
+   * @protected
+   * @override
+   */
+  teardownNode(node) {
+    if (this.__labelObserver) {
+      this.__labelObserver.disconnect();
+    }
+
+    // Default label is removed, do nothing.
+    if (node !== this.defaultNode) {
+      this.__applyDefaultLabel();
+    }
+
+    const hasLabel = this.__hasLabel(this.getSlotChild());
+    this.__toggleHasLabel(hasLabel);
+  }
+
+  /**
+   * Set label based on corresponding host property.
+   *
+   * @param {string} label
+   */
+  setLabel(label) {
+    this.label = label;
+
+    this.__updateDefaultLabel(label);
+  }
+
+  /**
+   * Set callback to be called when label changes.
+   *
+   * @param {Function} callback
+   */
+  setLabelChangedCallback(callback) {
+    this.labelChangedCallback = callback;
+  }
+
+  /**
+   * @param {HTMLElement} labelNode
+   * @private
+   */
+  __applyCustomLabel(labelNode) {
+    this.__updateLabelId(labelNode);
+    const hasLabel = this.__hasLabel(labelNode);
+    this.__toggleHasLabel(hasLabel);
+  }
+
+  /** @private */
+  __applyDefaultLabel() {
+    let labelNode = this.getSlotChild();
+
+    // Restore the default label element.
+    if (!labelNode) {
+      labelNode = this.defaultNode;
+      labelNode.id = this.defaultId;
+      this.__observeLabel(labelNode);
+      this.host.appendChild(labelNode);
+    }
+
+    this.__updateDefaultLabel(this.label);
+  }
+
+  /**
+   * @param {HTMLElement} labelNode
+   * @return {boolean}
+   * @private
+   */
+  __hasLabel(labelNode) {
+    if (!labelNode) {
+      return false;
+    }
+
+    return labelNode.children.length > 0 || this.__isNotEmpty(labelNode.textContent);
+  }
+
+  /**
+   * @param {string} label
+   * @private
+   */
+  __isNotEmpty(label) {
+    return Boolean(label && label.trim() !== '');
+  }
+
+  /**
+   * @param {HTMLElement} labelNode
+   * @private
+   */
+  __observeLabel(labelNode) {
+    this.__labelObserver = new MutationObserver((mutations) => {
+      mutations.forEach((mutation) => {
+        const target = mutation.target;
+
+        // Ensure the mutation target is the currently connected label
+        // to ignore async mutations dispatched for removed element.
+        const isLabelMutation = target === this.node;
+
+        if (mutation.type === 'attributes') {
+          // We use attributeFilter to only observe ID mutation,
+          // no need to check for attribute name separately.
+          if (isLabelMutation && target.id !== this.defaultId) {
```

Comment:
If the user sets the label node id to `this.defaultId` by hand for any reason, it won't update the `labelId` property. However, it would probably be better to make `labelId` a getter as this will solve the mentioned problem along the way.

Reply:
Changed to use a getter as you suggested.